packages/api/src/router/search-engine/search-engine-router.ts

```diff
@@ -74,8 +88,9 @@ export const searchEngineRouter = createTRPCRouter({
       .set({
         name: input.name,
         iconUrl: input.iconUrl,
-        urlTemplate: input.urlTemplate,
+        urlTemplate: "urlTemplate" in input ? input.urlTemplate : null,
         description: input.description,
+        integrationId: "integrationId" in input ? input.integrationId : null,
```

Comment:
Either we need to make the `type` changable as well or we need to make the `SegmentedControl` in the form readonly